src/lib/swapper/swappers/ThorchainSwapper/utils/getL1ToLongtailQuote.ts

```diff
@@ -0,0 +1,174 @@
+        code: TradeQuoteError.InternalError,
+        details: { sellAssetChainId: sellChainId },
+      }),
+    )
+  }
+
+  // TODO: Move these constants outside
+  // TODO2: use more than just UniswapV3, and also consider trianglar routes.
+  const ALLOWANCE_CONTRACT = '0xF892Fef9dA200d9E84c9b0647ecFF0F34633aBe8' // TSAggregatorTokenTransferProxy
+
+  const l1Tol1QuoteInput: GetTradeQuoteInput = {
+    ...input,
+    buyAsset: nativeBuyAsset,
+    sellAsset,
+    sellAmountIncludingProtocolFeesCryptoBaseUnit: sellAmountCryptoBaseUnit,
+  }
+
+  const thorchainQuotes = await getL1quote(
+    l1Tol1QuoteInput,
+    streamingInterval,
+    TradeType.L1ToLongTail,
+  )
+
+  const unwrappedThorchainQuotes = thorchainQuotes.unwrap()
+
+  let bestAggregator: AggregatorContract
+  let quotedAmountOut: bigint
+
+  const promises = await Promise.allSettled(
+    unwrappedThorchainQuotes.map(async quote => {
+      const onlyStep = quote.steps[0]
+
+      const maybeBestAggregator = await getBestAggregator(
+        nativeBuyAsset,
+        getWrappedToken(nativeBuyAsset),
+        getTokenFromAsset(buyAsset),
+        onlyStep.buyAmountAfterFeesCryptoBaseUnit,
+      )
+
+      if (maybeBestAggregator.isErr()) return Err(maybeBestAggregator.unwrapErr())
+
+      const unwrappedResult = maybeBestAggregator.unwrap()
+
+      bestAggregator = unwrappedResult.bestAggregator
+      quotedAmountOut = unwrappedResult.quotedAmountOut
+
+      const updatedMemo = addAggregatorAndDestinationToMemo({
+        aggregator: bestAggregator,
+        destinationToken: fromAssetId(buyAsset.assetId).assetReference as Address,
+        minAmountOut: convertDecimalPercentageToBasisPoints(
+          slippageTolerancePercentageDecimal ??
+            getDefaultSlippageDecimalPercentageForSwapper(SwapperName.Thorchain),
+        ).toString(),
+        quotedMemo: quote.memo,
+      })
+
+      const { data, router, vault } = await getEvmThorTxInfo({
+        sellAsset,
+        sellAmountCryptoBaseUnit,
+        memo: updatedMemo,
+        expiry: quote.expiry,
+      })
+
+      return Ok({
+        ...quote,
+        memo: updatedMemo,
+        data,
+        router,
+        vault,
+        aggregator: bestAggregator,
+        steps: quote.steps.map(s => ({
+          ...s,
+          buyAsset,
+          buyAmountAfterFeesCryptoBaseUnit: quotedAmountOut.toString(),
+          // This is wrong, we should get the get the value before fees or display ETH value received after the thorchain bridge
+          buyAmountBeforeFeesCryptoBaseUnit: quotedAmountOut.toString(),
+          allowanceContract: ALLOWANCE_CONTRACT,
+        })) as MultiHopTradeQuoteSteps, // assuming multi-hop quote steps here since we're mapping over quote steps,
+        isLongtail: true,
+        longtailData: {
+          L1ToLongtailExpectedAmountOut: quotedAmountOut,
+        },
+      })
+    }),
+  )
+
+  if (promises.every(promise => isRejected(promise) || isErrorPromise(promise))) {
```

Comment:
I added a `isErrorPromise` type guard so we handle this properly in a monadic way, if not it could throw